src/crimson/osd/osd.cc

```diff
@@ -1402,6 +1403,93 @@ seastar::future<> OSD::send_beacon()
+{
+  LOG_PREFIX(OSD::run_bench);
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> dis(0, 255);
+  seastar::sstring collection_name = "run_bench";
+  coll_t collection;
+  DEBUG(" count={}, bsize={}, osize={}, onum={}", count, bsize, osize, onum);
+  return seastar::do_for_each(boost::make_counting_iterator<int64_t>(0),
+  boost::make_counting_iterator<int64_t>(onum),
+  [=, &collection](int64_t i) {
+    std::string data(osize, 'a');
+    ceph::buffer::list bl(osize);
+    bl.append(data);
+
+    std::string oid_str = fmt::format("disk_bw_test_{}", i);
+    ghobject_t oid(hobject_t(sobject_t(object_t(oid_str), 0)),
+                                       ghobject_t::NO_GEN, 
+                                       shard_id_t::NO_SHARD);
+    ceph::os::Transaction t;
+    t.write(collection, oid, 0, data.size(), bl);
+    ceph::os::Transaction cleanup;
+    cleanup.remove(collection, oid);
+    return seastar::make_ready_future<>();
+  }).then([=, this, &collection] {
+    return store.get_sharded_store().open_collection(collection).then(
+      [] (auto collection_ref) {
+      return collection_ref->flush_commit().then([=] (bool) {
+        return seastar::make_ready_future<>();
+      });
+    });
+  }).then([=, this, &elapsed, &ss, &collection]() mutable {
+    seastar::temporary_buffer<char> cleanupt_buf(0);
+    seastar::sstring cleanupt_oid;
+
+    auto start = std::chrono::steady_clock::now();
+    return seastar::do_for_each(boost::make_counting_iterator<int64_t>(0),
+      boost::make_counting_iterator<int64_t>(count / bsize),
+      [&collection, bsize, osize, onum, &gen, &dis](int64_t i) {
+      std::string data(bsize, 'a');
+      ceph::buffer::list bl;
+      bl.append(data);
+      std::string oid_str;
+      uint64_t offset = 0;
+      if (onum && osize) {
+        oid_str = fmt::format("disk_bw_test_{}", (int)(dis(gen) % onum));
+        offset = (dis(gen) % (osize / bsize)) * bsize;
+      } else {
+        oid_str = fmt::format("disk_bw_test_{}", i * bsize);
+      }
+      ghobject_t oid(hobject_t(sobject_t(object_t(oid_str), 0)),
+                                         ghobject_t::NO_GEN,
+                                         shard_id_t::NO_SHARD);
+      ceph::os::Transaction t;
+      t.write(collection, oid, offset, bsize, bl);
+      ceph::os::Transaction cleanup;
+      cleanup.remove(collection, oid);
+      bl.clear();
+      return seastar::make_ready_future<>();
+    }).then([=, this, &collection]() {
+      return store.get_sharded_store().open_collection(collection).then(
+        [] (auto collection_ref) {
+        return collection_ref->flush_commit().then([=] (bool) {
+          return seastar::make_ready_future<>();
+        });
+      });
+    }).then([&elapsed, start]() {
+      seastar::timer<>::time_point end = std::chrono::steady_clock::now();
+      *elapsed = std::chrono::duration<double>(end - start).count();
```

Comment:
We should wait for the transactions to take place for measuring the elapsed time